torch/distributed/_tensor/_utils.py

```diff
@@ -224,3 +224,17 @@ def compute_local_stride(
     return tuple(
         global_stride[i] // stride_divisors[i] for i in range(len(global_stride))
     )
+
+
+def _disable_dynamo(fn=None, recursive=True):
+    """
+    This util is merely a wrapper on top of torch._disable_dynamo. torch._disable_dynamo
+    uses funtools wrap where it incurs high CPU overhead. This util is used to
+    avoid circular imports issues, and early exit if it's not in compiling.
+    """
+    import torch.compiler
+
+    if not torch.compiler.is_compiling():
```

Comment:
Hmm… I’m not sure if this will do what we want 🤔 

I think is_compiling will be evaluated once at import time (when DTensor) is imported, and no-op since dynamo isn’t enabled?

Reply:
:( yeah I figured the test failures are related to this.

I'm not sure if there're any ways to approach this..